src/account_sim.py

```diff
@@ -17,15 +18,16 @@ class AccountSimulator(IAccount):
     positions_df: pd.DataFrame
     positions: dict
 
-    def __init__(self, balance, profit) -> None:
+    def __init__(self, symbol, balance, profit) -> None:
         self.balance = balance
         self.profit = profit
         self.ticket = 500000000
         self.positions = dict()
-        self.positions_df = pd.DataFrame('symbol','ticket','time','type','volume','price','current_price','profit') 
+        self.positions_df = pd.DataFrame(columns=['symbol','ticket','time','type','volume','price','current_price','profit'])
+        self.symbol = symbol
 
-    def update_balance(self, profit) -> None:
-        self.balance += profit
+    def update_balance(self, capital_committed) -> None:
```

Comment:
I believe this implies that anytime a trade gets made in the simulator, it behaves as though it successfully trades right away.

In the actual mt5 trading mechanism, we send a trade and mt5 trades only when the conditions are right. If the conditions do not become right by the time we ask for another trade, then the trade is closed as a part of the close_all_positions method.

It is much more straightforward to implement only the situation where the trade is guaranteed to go through. But if the trade always goes through we should know that in any testing we do with the sim bot that we will miss the scenario of cancelling orders unless we explicitly write another code path to test that out.

Reply:
Basically, if I understand the mock correctly, we will miss hitting remove_position and update_position until we have a way to test out these scenarios.

This isn't really a problem but more of a TODO for future us when we need to test out these scenarios.

Reply:
I agree with this comment, in a later implementation of this sim class I was planning to have trades fail a certain % of the time. It is on my list of TODOs